src/status.ts

```diff
@@ -0,0 +1,267 @@
+          title: "Restart Server",
+          command: ServerCommand.Error,
+        };
+      }
+    }
+  }
+
+  public async installGems(): Promise<boolean> {
+    if (await bundleCheck()) {
+      this.updateStatus(ServerCommand.Error);
+      const status: vscode.LanguageStatusItem = this.createStatusItem(
+        "installGems",
+        "Ruby LSP: The gems in the bundle are not installed.",
+        vscode.LanguageStatusSeverity?.Error
+      );
+
+      this.context.subscriptions.push(
+        vscode.commands.registerCommand("ruby-lsp.installGems", () => {
+          status.text = "Ruby LSP: Installing gems...";
+          status.command = undefined;
+          status.busy = true;
+          bundleInstall()
+            .then(() => {
+              status.dispose();
+              this.addMissingGem();
+            })
+            .catch(() => {
+              status.dispose();
+              this.createStatusItem(
+                "installFail",
+                "Ruby LSP: Failed to install gems.",
+                vscode.LanguageStatusSeverity?.Error
+              );
+            });
+        })
+      );
+
+      status.command = {
+        title: "Run bundle install",
+        command: "ruby-lsp.installGems",
+      };
+      return true;
+    }
+    return false;
+  }
+
+  public async addMissingGem(): Promise<boolean> {
+    if (await isGemMissing()) {
+      this.updateStatus(ServerCommand.Error);
+      const status: vscode.LanguageStatusItem = this.createStatusItem(
+        "addMissingGem",
+        "Ruby LSP: Bundle Add",
+        vscode.LanguageStatusSeverity?.Error
+      );
+
+      this.context.subscriptions.push(
+        vscode.commands.registerCommand("ruby-lsp.addMissingGem", () => {
+          status.text = "Ruby LSP: Adding gem...";
+          status.command = undefined;
+          status.busy = true;
+          addGem()
+            .then(() => status.dispose())
+            .catch(() => {});
+        })
+      );
+
+      status.command = {
+        title: "Run bundle add and install",
+        command: "ruby-lsp.addMissingGem",
+      };
+      return true;
+    }
+    return false;
+  }
+
+  private async activateGemOutdatedButton() {
+    const gemOutdated = await isGemOutdated();
+    if (!gemOutdated) {
+      return;
+    }
+
+    const commandId = "updateOutdatedGem";
+
+    const status: vscode.LanguageStatusItem = this.createStatusItem(
+      "updateGem",
+      "Ruby LSP: The gem is not up-to-date",
+      vscode.LanguageStatusSeverity?.Warning
+    );
+
+    this.context.subscriptions.push(
+      vscode.commands.registerCommand(commandId, async () => {
+        status.text = "Ruby LSP: Updating Ruby LSP";
+        status.busy = true;
+        status.command = undefined;
+
+        const result = await updateGem();
+
+        if (result.stderr.length > 0) {
+          status.text = "Ruby LSP: Update failed";
+          status.command = {
+            title: "Try again",
```

Comment:
Is there a scenario in which trying again can succeed? I feel like if bundle update fails, the dev is probably going to have to jump to the terminal and figure it out.